pkg/session_recording/recordings.jsx

```diff
@@ -95,33 +123,72 @@
         constructor(props) {
             super(props);
             this.handleDateChange = this.handleDateChange.bind(this);
+            this.clearField = this.clearField.bind(this);
+            this.markDateField = this.markDateField.bind(this);
+            this.state = {
+                invalid: false,
+                date: this.props.date,
```

Comment:
How about we keep both the last valid parsed date and the current date string here, in state? Then we wouldn't have to keep it in View. Then, I think, the change callback supplied by the component user could be invoked with both of them too. Let's say first the last parsed date (null if none), then the original string. This way View would be able to update both the URL and the filtering.

This way also we wouldn't have to expose all the parsing intricacies, and the parseDate function in particular, to the component user. What do you think?

Reply:
I definitely like the idea, because it's proper way in terms of not having too much unnecessary binding between different classes/components. I'll try to implement it this way.